Added some initial util classes

f2rf2r opened the pull request.

Commit:
Added some initial util classes

ryanmeador commented on commonutils/src/main/java/io/intrepid/commonutils/IntrepidJavaUtils.java:
```diff
@@ -0,0 +1,59 @@
+        for (int j = 0; j < bytes.length; j++) {
+            int v = bytes[j] & 0xFF;
+            hexChars[j * 2] = hexArray[v >>> 4];
+            hexChars[j * 2 + 1] = hexArray[v & 0x0F];
```
Does this have a significant performance advantage over String.format()?  If performance is the reason for it being so opaque, I think there are further improvements we can make.

f2rf2r replied:
It wasn't for the performance reasons, I just haven't thought about using String.format(). I'm fine with going with that approach if you think it's more readable.

russhwolf commented on commonutils/src/main/java/io/intrepid/commonutils/IntrepidAndroidUtils.java:
```diff
@@ -0,0 +1,63 @@
+    public static float convertDpToPixel(@NonNull Context context, float dp) {
+        Resources resources = context.getResources();
+        DisplayMetrics metrics = resources.getDisplayMetrics();
+        return dp * (metrics.densityDpi / 160f);
```
There are APIs in `TypedValue` to do this sort of thing.
`TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, metrics);`

Commit:
PR comment changes
- Use to build-in TypedValue.applyDimension() for converting dp to px
- Use String.format() for converting byte array to hex string

Commit:
Use Robolectric for some of Android tests

Commit:
Improve maven publishing and include the source jar

streetsofboston commented on commonutils/publish.gradle:
```diff
@@ -0,0 +1,81 @@
+    return property == null ? "" : property
+}
+
+ext.artifactPath = "build/outputs/aar/commonutils-release.aar"
```
Is this 'hard-coded' path robust enough? Could there be an already existing variable/field holding (part of) this path.

f2rf2r replied:
Fixed

streetsofboston commented on commonutils/src/main/java/io/intrepid/commonutils/IntrepidJavaUtils.java:
```diff
@@ -0,0 +1,56 @@
+public class IntrepidJavaUtils {
+
+    @NonNull
+    public static String bytesToHexString(@NonNull byte[] bytes) {
```
Is this big or little endian?
If big-endian, consider using BigInteger instead: 
https://developer.android.com/reference/java/math/BigInteger.html#BigInteger(int, byte[])
and then calling
https://developer.android.com/reference/java/math/BigInteger.html#toString(int)

f2rf2r replied:
I tried using the BigInteger version, but its result doesn't include the leading 0.

streetsofboston replied:
I see.
Just be sure to add some java-doc stating the byte-order of the `bytes` array.

Commit:
Update publish.gradle to be more flexible regarding aar path

streetsofboston commented:
Ship it.

gdaniels commented:
:shipit:
